security/xmlsec-tests.py

```diff
@@ -43,21 +43,34 @@ def setUp(self):
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/lsh123/xmlsec/archive/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'xmlsec-master')
-        os.chdir(self.sourcedir)
-        process.run('./autogen.sh', ignore_status=True)
-        build.make(self.sourcedir)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = "https://github.com/lsh123/xmlsec/archive/master.zip"
+            url = self.params.get('url', default=def_url)
+            tarball = self.fetch_asset(url, expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.srcdir = os.path.join(self.workdir, 'xmlsec-master')
+            os.chdir(self.srcdir)
+            process.run('./autogen.sh', ignore_status=True)
+            if build.make(self.srcdir, process_kwargs={"ignore_status": True}):
```

Comment:
Just wanted to understand why ignore_status is True

Reply:
@abdhaleegit
To handle 'make' error gracefully.
```
if build.make(self.srcdir, process_kwargs={"ignore_status": True}):
    self.fail("'make' command failed.")
```
If not the error from build.make will look very odd.

Reply:
Ok